### ISSUE TITLE
vcd_vapp_vm: network: Update of ip_allocation_mode does not work from POOL to MANUAL if you keep the same ip

### ISSUE THREAD

**Issue by jpbuecken** [vmware/terraform-provider-vcd#552]
### Version
Used a self compiled version based on #543.

### Affected Resource(s)
Please list the resources as a list, for example:
- vcd_vapp_vm

### Terraform Configuration Files
```hcl
  network {
    type               = "org"
    name               = "NETWORK1"
    ip_allocation_mode = "POOL"
  }
```

Build a VM with above network hcl. 
Change the above HCL to 

```hcl
  network {
    type               = "org"
    name               = "NETWORK1"
    ip_allocation_mode = "MANUAL"
    ip                      = "<use the same IP that has been applied by POOL"
  }
```
### Expected Behavior
If you apply the new hcl, the ip_allocation_mode should change to MANUAL, with the same ip that has been applied earlier.

### Actual Behavior
The ip allocation mode is not changed. Every terraform appky shows the change again:
```
      ~ network {
            adapter_type       = "VMXNET3"
            connected          = true
            ip                 = "172.16.250.8"
          ~ ip_allocation_mode = "POOL" -> "MANUAL"
            is_primary         = true
            mac                = "00:50:56:01:00:d4"
            name               = "NETWORK2"
            type               = "org"
        }
```
If you change the IP to another value than the current one, it is working:

```
      ~ network {
            adapter_type       = "VMXNET3"
            connected          = true
            ip                 = "172.16.250.8" -> "172.16.250.9"
          ~ ip_allocation_mode = "POOL" -> "MANUAL"
            is_primary         = true
            mac                = "00:50:56:01:00:d4"
            name               = "NETWORK2"
            type               = "org"
        }
```
--->>> OK!

**Comment by jpbuecken** [comment]
Hello,
this is still an issue with VCD 10.4 and provider 3.8.
But the UI has the same issue. 

Maybe both, terraform and UI uses a wrong API call, or this is a VCD issue.

If you want to reach out to VCD engineering and need a reference, our VMware Case is SR 22384690011 .

**Comment by jpbuecken** [comment]
Anyway, the terraform provider may be able to work around this issue.
Idea: On update, first send an API call to set network ip_allocation_mode from pool to dhcp, then send the original API call to configure the interface with the static ip.

In case you follow the idea, please do not remove the network name during update (e.g. do not set it to none), this will lead to connection interruption within the OS.

**Comment by jpbuecken** [comment]
Hello,
this should be fixed with 10.4.2

From the release notes:
https://docs.vmware.com/en/VMware-Cloud-Director/10.4.2/rn/vmware-cloud-director-1042-release-notes/index.html
```
Updating the VM NIC from Static-Pool to Static-Manual IP mode does not apply the change

In a VM, if the VM NIC is connected to a vApp direct network and you attempt to update the IP mode from Static-Pool to Static-Manual while keeping the same IP address, the system does not save the change and the configuration of the IP mode remains Static-Pool.
```

So I close my request for the provider side here.
